impeller/core/vertex_buffer.h

```diff
@@ -27,18 +30,11 @@ struct VertexBuffer {
   /// The type of indices in the index buffer. The indices must be tightly
   /// packed in the index buffer.
   ///
-  IndexType index_type = IndexType::kUnknown;
+  IndexType index_type = IndexType::kNone;
```

Comment:
Why change this one?

Reply:
It seems like a better default to assume there is no index buffer until one is set with `pass.SetIndexBuffer`. Otherwise stuff that doesn't want to use an index buffer will need to call `pass.SetIndexBuffer({}, IndexBuffer.kNone)`

Reply:
Although I guess in practice, stuff that doesn't use an index buffer will need to call `pass.SetIndexBuffer({}, IndexBuffer.kNone)` anyway, since it would otherwise be making assumptions about the previous draw call.

Reply:
Oh in flutter_gpu? I think you should actually use an entirely different struct. Lets maybe pretend this one doesn't exist since its completely tied to how all the existing contents/geometries work.

Reply:
Nah I meant in Impeller in general. I'm actually thinking we should remove `bool BindVertices(VertexBuffer buffer)` entirely and disallow setting the vertex/index buffers using this struct everywhere. I changed this back to `kUnknown` for now.

Reply:
Yeah see comment above. i agree with removing it (later) for now I would not further change the struct and instead update flutter_gpu to use the other render pass APIs instead.